### ISSUE TITLE
`okta_app_basic_auth` optional fields are failing as required

### ISSUE THREAD

**Issue by bryantbiggs** [okta/terraform-provider-okta#223]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.13.4

### Affected Resource(s)

* `okta_app_basic_auth` resource

### Terraform Configuration Files

```hcl
resource "okta_app_basic_auth" "service_account" {
  label = "A Service Account"
}
```

### Error Output

```
Error: The API returned an error: Api validation failed: url. Causes: errorSummary: url: The field cannot be left blank, errorSummary: authURL: The field cannot be left blank

  on apps.tf line 289, in resource "okta_app_basic_auth" "service_account":
 289: resource "okta_app_basic_auth" "service_account" {
```

### Expected Behavior

The resource should be able to be created without `url` or `auth_url` as they are marked as optional.

### Actual Behavior

Resource fails to create stating that  `url` and `authUrl` are required

### Steps to Reproduce

1. Copy resource above 
2. `terraform apply`

### Important Factoids

No

### References

<img width="526" alt="image" src="https://user-images.githubusercontent.com/10913471/100100573-065c8780-2e2f-11eb-9603-c784bb7c42d4.png">

<img width="425" alt="image" src="https://user-images.githubusercontent.com/10913471/100100721-402d8e00-2e2f-11eb-8cd2-106562e10bca.png">

**Comment by bogdanprodan-okta** [comment]
Hi @bryantbiggs! Thanks for creating this issue! I'll make an investigation and create PR to fix this if necessary.

I made a test call to create a simple app without these values. As it turned out, these are required parameters.
Request:
```
curl --location --request POST 'https://my-domain.okta.com/api/v1/apps' \
--header 'Authorization: SSWS *****' \
--header 'Content-Type: application/json' \
--data-raw '{
    "name": "template_basic_auth",
    "label": "Example",
    "signOnMode": "BASIC_AUTH",
    "settings": {
        "app": {}
    }
}'
```
Response:
```
{
    "errorCode": "E0000001",
    "errorSummary": "Api validation failed: url",
    "errorLink": "E0000001",
    "errorId": "oaeAHnXl-RxQ0ibBNtgOaWCyg",
    "errorCauses": [
        {
            "errorSummary": "url: The field cannot be left blank"
        },
        {
            "errorSummary": "authURL: The field cannot be left blank"
        }
    ]
}
```
This will be fixed in #220. Cheers!

**Comment by bryantbiggs** [comment]
thanks @bogdanprodan-okta !

**Comment by bryantbiggs** [comment]
closed with #220
